Refactor document status and menubar padding

mejo- opened the pull request:
### 📝 Summary

* fix: Make CollisionResolveDialog and DocumentStatus sticky
* fix: Remove overflow property from editor wrapper
* refactor(css): Use default-grid-baseline for menubar padding
* fix(Editor): Add property `lock` to data
* test(cy): Add regression test for sticky elements

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests

mejo- commented:
/backport to stable31

mejo- commented:
I again did extensive testing to find possible regressions and added another fix concerning mobile view with menubar at the bottom.

 This is the test matrix I went through:

✔️ = no problems
❌✅ = problem, but fixed now

| app         | view    | content | document status | collision | status |
| ----------- | ------- | ------- | --------------- | --------- | ------ |
| collectives | desktop | long    | no              | no        | ✔️      |
| collectives | desktop | long    | yes             | no        | ✔️      |
| collectives | desktop | long    | yes             | yes       | ✔️      |
| collectives | desktop | long    | no              | no        | ✔️      |
| collectives | desktop | short   | no              | no        | ✔️      |
| collectives | desktop | short   | yes             | no        | ✔️      |
| collectives | desktop | short   | yes             | yes       | ✔️      |
| collectives | desktop | long    | no              | no        | ✔️      |
| collectives | mobile  | long    | no              | no        | ❌✅ |
| collectives | mobile  | long    | yes             | no        | ✔️      |
| collectives | mobile  | long    | yes             | yes       | ✔️      |
| collectives | desktop | long    | no              | no        | ✔️      |
| collectives | mobile  | short   | no              | no        | ✔️ |
| collectives | mobile  | short   | yes             | no        | ✔️      |
| collectives | mobile  | short   | yes             | yes       | ✔️      |
| collectives | desktop | long    | no              | no        | ✔️      |
| standalone  | desktop | long    | no              | no        | ✔️      |
| standalone  | desktop | long    | yes             | no        | ❌✅   |
| standalone  | desktop | long    | yes             | yes       | ❌✅   |
| collectives | desktop | long    | no              | no        | ✔️      |
| standalone  | desktop | short   | no              | no        | ✔️      |
| standalone  | desktop | short   | yes             | no        | ✔️      |
| standalone  | desktop | short   | yes             | yes       | ✔️      |
| collectives | desktop | long    | no              | no        | ✔️      |
| standalone  | mobile  | long    | no              | no        | ✔️      |
| standalone  | mobile  | long    | yes             | no        | ✔️      |
| standalone  | mobile  | long    | yes             | yes       | ❌✅   |
| collectives | desktop | long    | no              | no        | ✔️      |
| standalone  | mobile  | short   | no              | no        | ✔️      |
| standalone  | mobile  | short   | yes             | no        | ✔️      |
| standalone  | mobile  | short   | yes             | yes       | ✔️      |

max-nextcloud commented:
## Scrollbar on mobile

I'm now seeing a scrollbar on short documents in chromium and if a message is displayed the scrollable area becomes higher:

https://github.com/user-attachments/assets/bb7f93ea-a63c-4d34-b361-62dc13a7108f

## Positioning on standalone (Chromium)

*(Not sure this is an issue...)*
With a short document the status now shows in the middle of the doc:

![Bildschirmfoto vom 2025-04-30 08-48-30](https://github.com/user-attachments/assets/3d84e780-678c-4ed6-a05e-d047a745b3de)

mejo- commented:
>  With a short document the status now shows in the middle of the doc:

Fixed this by using flexbox layout and `flex-grow: 1`.

> I'm now seeing a scrollbar on short documents in chromium and if a message is displayed the scrollable area becomes higher:

Both also fixed now.
